capstone/src/main/webapp/script2.js

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var data_for_select_field ={};
+//the data_for_autocomplete represents what will be shown on the form:
+//the keys of this object is the text, while the value is an image url of the associated text.
+//I'm setting the values as null because an image is not required in our case
+var data_for_autocomplete = {};
+
+async function getUniversity() {
+    const response = await fetch("/university");
+    const result = await response.json();
+    setUpAutoComplete(result);
+}
+
+function setUpAutoComplete(university_data_from_request){
+    for(el in university_data_from_request) {
+        var university = university_data_from_request[el];
+        var university_description = university["name"] + ", " + university["state"];
+        data_for_select_field[university["id"]] = university_description;
+        data_for_autocomplete[university_description] = null;
+    }
+}
+
+getUniversity();
+
+document.addEventListener('DOMContentLoaded', function() {
+    //Init auto complete.
+    const songListIns = (function(){
+    const acElms = document.querySelectorAll('.autocomplete');
+    const instances = M.Autocomplete.init(acElms, {
+        data: data_for_autocomplete
+    })
+    return instances[0]
+    })()
+})
+
+function checkUniversityValue() {
+    var val = document.getElementById("university").value;
+    var error = document.getElementById("university_error");
+    error.innerHTML = "";
+    for ( university_id in data_for_select_field ){
+        if (data_for_select_field[university_id] == val) {
+            return university_id;
+        }
+    }
+    const pElement = document.createElement("p");
+    pElement.style.color = "red";
+    pElement.innerHTML = "Your choice is not valid";
+    error.appendChild(pElement);
+    return ;
+}
+
+function checkFullNameInputValue(){
+    var val = document.getElementById("full_name").value;
+    var error = document.getElementById("full_name_error");
+    error.innerHTML = "";
+    if (val == "" || val == " "){
```

Comment:
What about 2+ spaces or a bunch of tabs? Try https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim